test/e2e/snat/snat_suite_test.go

```diff
@@ -0,0 +1,153 @@
+	Expect(len(nodes.Items)).Should(BeNumerically(">", 1))
+
+	// Set the primary node for testing
+	primaryNodeInPublicSubnet = nodes.Items[0]
+
+	By("Getting Public and Private subnets")
+	vpcConfig, err := utils.GetClusterVPCConfig(f)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(len(vpcConfig.PublicSubnetList)).To(BeNumerically(">", 0))
+	Expect(len(vpcConfig.PrivateSubnetList)).To(BeNumerically(">", 0))
+
+	msg := fmt.Sprintf("Creating a keyPair with name: %s if it doesn't exist", DEFAULT_KEY_PAIR)
+	By(msg)
+	keyPairOutput, _ := f.CloudServices.EC2().DescribeKey(DEFAULT_KEY_PAIR)
+
+	exists := false
+	if keyPairOutput != nil {
+		for _, keyPair := range keyPairOutput.KeyPairs {
+			if *keyPair.KeyName == DEFAULT_KEY_PAIR {
+				exists = true
+				break
+			}
+		}
+	}
+
+	if exists {
+		fmt.Println("KeyPair already exists")
+	} else {
+		fmt.Println("KeyPair doesn't exist, will be created")
+		_, err := f.CloudServices.EC2().CreateKey(DEFAULT_KEY_PAIR)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	privateSubnetId = vpcConfig.PrivateSubnetList[0]
+
+	By("Getting Cluster Security Group Id")
+	out, err := f.CloudServices.EKS().DescribeCluster(f.Options.ClusterName)
+	Expect(err).NotTo(HaveOccurred())
+
+	clusterSecurityGroupId := out.Cluster.ResourcesVpcConfig.ClusterSecurityGroupId
+
+	msg = fmt.Sprintf("Deploying a self managed nodegroup of size 1 in private subnet %s", privateSubnetId)
+	By(msg)
+	props = utils.NodeGroupProperties{
+		NgLabelKey:    "test-label-key",
+		NgLabelVal:    "test-label-val",
+		AsgSize:       1,
+		NodeGroupName: "snat-test-ng",
+		Subnet: []string{
+			privateSubnetId,
+		},
+		InstanceType: "m5.large",
+		KeyPairName:  DEFAULT_KEY_PAIR,
+	}
+
+	err = utils.CreateAndWaitTillSelfManagedNGReady(f, props)
+	Expect(err).NotTo(HaveOccurred())
+
+	nodeList, err := f.K8sResourceManagers.NodeManager().GetNodes(props.NgLabelKey,
+		props.NgLabelVal)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(len(nodeList.Items)).Should(BeNumerically(">", 0))
+
+	// Get ref to the only node from newly created nodegroup
+	primaryNodeInPrivateSubnet = nodeList.Items[0]
+
+	By("Fetching existing Security Groups from the newly created node group instance")
+	instanceOutput, err := f.CloudServices.EC2().DescribeInstancesWithFilters(map[*string][]*string{
+		aws.String("private-dns-name"): {
+			aws.String(primaryNodeInPrivateSubnet.Name),
+		},
+	})
+
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(instanceOutput.Reservations)).To(BeNumerically(">", 0))
+	Expect(len(instanceOutput.Reservations[0].Instances)).To(BeNumerically(">", 0))
+
+	instance := instanceOutput.Reservations[0].Instances[0]
+
+	existingSecurityGroups := instance.SecurityGroups
+	networkInterfaceId := f.CloudServices.EC2().GetPrimaryNetworkInterfaceId(instance.NetworkInterfaces, instance.PrivateIpAddress)
+	Expect(networkInterfaceId).NotTo(Equal(BeNil()))
+
+	securityGroupIds := make([]*string, 0, len(existingSecurityGroups)+1)
```

Comment:
We are using following template: https://github.com/awslabs/amazon-eks-ami/blob/master/amazon-eks-nodegroup.yaml which creates a self managed node group. We will have to modify the template so that it creates managed node group instead and attaches the ClusterSecurityGroupId. 
Ref: https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html . Since I was using the existing method to create self managed node, I had to take this approach of attaching the missing SG after its creation.

Reply:
I see there's no parameter for passing additional SG.